kit/azure/networking/resources.network.tf

```diff
@@ -0,0 +1,282 @@
+  virtual_network_name = azurerm_virtual_network.hub_network.name
+  address_prefixes     = [cidrsubnet(var.address_space, 2, 1)]
+
+  service_endpoints = [
+    "Microsoft.Storage",
+  ]
+}
+
+resource "azurerm_subnet" "mgmt" {
+  name                 = "Management"
+  resource_group_name  = azurerm_resource_group.hub_resource_group.name
+  virtual_network_name = azurerm_virtual_network.hub_network.name
+  address_prefixes     = [cidrsubnet(var.address_space, 2, 3)]
+
+  service_endpoints = [
+    "Microsoft.Storage",
+  ]
+}
+
+resource "azurerm_network_security_group" "mgmt" {
+  name                = "subnet-mgmt-nsg"
+  location            = azurerm_resource_group.hub_resource_group.location
+  resource_group_name = azurerm_resource_group.hub_resource_group.name
+}
+
+resource "azurerm_subnet_network_security_group_association" "mgmt" {
+  subnet_id                 = azurerm_subnet.mgmt.id
+  network_security_group_id = azurerm_network_security_group.mgmt.id
+}
+
+resource "azurerm_network_watcher_flow_log" "mgmt_logs" {
+  count = var.netwatcher != null ? 1 : 0
+
+  network_watcher_name      = azurerm_network_watcher.netwatcher[0].name
+  resource_group_name       = azurerm_resource_group.netwatcher[0].name
+  name                      = "${azurerm_resource_group.hub_resource_group.name}-subnet-mgmt-nsg"
+  network_security_group_id = azurerm_network_security_group.mgmt.id
+  storage_account_id        = azurerm_storage_account.flowlogs.id
+  enabled                   = true
+  version                   = 2
+
+  traffic_analytics {
+    enabled               = true
+    workspace_id          = var.netwatcher.log_analytics_workspace_id_short
+    workspace_region      = var.location
+    workspace_resource_id = var.netwatcher.log_analytics_workspace_id
+  }
+
+  retention_policy {
+    days    = 0
+    enabled = true
+  }
+}
+
+data "azurerm_monitor_diagnostic_categories" "mgmt" {
+  resource_id = azurerm_network_security_group.mgmt.id
+}
+
+resource "azurerm_monitor_diagnostic_setting" "mgmt" {
+  count = var.diagnostics != null ? 1 : 0
+
+  name                           = "mgmt-nsg-diag"
+  target_resource_id             = azurerm_network_security_group.mgmt.id
+  log_analytics_workspace_id     = local.parsed_diag.log_analytics_id
+  eventhub_authorization_rule_id = local.parsed_diag.event_hub_auth_id
+  storage_account_id             = local.parsed_diag.storage_account_id
+  dynamic "enabled_log" {
+    for_each = data.azurerm_monitor_diagnostic_categories.mgmt.log_category_types
+    content {
+      category = enabled_log.value
+    }
+  }
+}
+
+resource "azurerm_network_security_rule" "mgmt" {
+  count = length(local.merged_mgmt_nsg_rules)
+
+  resource_group_name                        = azurerm_resource_group.hub_resource_group.name
+  network_security_group_name                = azurerm_network_security_group.mgmt.name
+  priority                                   = 100 + 100 * count.index
```

Comment:
d: This means the index of the rule in local.merged_mgmt_nsg_rules determines the priority.   

They are defined via a [merge](https://developer.hashicorp.com/terraform/language/functions/merge)
```hcl
merged_mgmt_nsg_rules = flatten([
    for nsg in var.management_nsg_rules : merge(nsg)
  ])
```

This should be made clear.

I couldn't find out if the order is determined deterministically. It's not documented on the merge function documentation page. And it's not stated in the README.

Reply:
But it still means that priority is determined by the index in some list.

So for [a, b] means 
- a has prio 100 and 
- b has prio 200.

However if you do [b, a] you get the reverse: 
- a has prio 200 and 
- b has prio 100.

If users know that the position of the rule determines the prio, I'm fine with it. But this is not documented

Reply:
Good point, u are right. At the end its a filled pre-configuration should point out the possibilities and be adapted for further processing. I will adding a comment, thanks.